app/controllers/messages_controller.rb

```diff
@@ -0,0 +1,29 @@
+class MessagesController < ApplicationController
+  before_action :authenticate_user!
+  
+  def index
+    @users = User.near([current_user.latitude, current_user.longitude], 2000)
+    @messages = Message.all
+  end
+  
+  def create
+    if Message.includes(current_user.id)
+      @message = Message.between(params[:sender_id], params[:recipient_id]).first
+      redirect_to message_contents_path(@message)
+    else
+      @message = Message.create(message_params)
+      redirect_to message_contents_path(@message)
+    end
+  end
+  
+  def destroy
+    message = Message.find(params[:message_id])
+    message.destroy
+    redirect_to message_path(message)
+  end
+  
+private
+  def message_params
+    params.permit(:sender_id, :recipient_id)
```

Comment:
You're missing the `.require` part of this statement.

Reply:
params.require(:message).permit(:sender_id, :recipient_id)

param is missing or the value is empty: message

I added the require before, but it caused error. Thus, I did it without it.

Reply:
You *need* to included the `.require` part in order for this to actually use strong parameters (which you should definitely be using). 

Please take some time to figure out how you can make this work with `.require`. The problem might be that you don't have a `new` method in your controller.